sdk/identity/azure-identity/CHANGELOG.md

```diff
@@ -8,6 +8,8 @@
 
 ### Bugs Fixed
 
+- Fixed pyright error for `MsalCredential` ([#35184](https://github.com/Azure/azure-sdk-for-python/pull/35184))
```

Comment:
Can try to avoid using internal credential names and keep it more general.

```suggestion
- Fixed typing errors when certain credentials are used as context managers. ([#35415](https://github.com/Azure/azure-sdk-for-python/pull/35415))
```